controllers/inventory-controller.js

```diff
@@ -69,7 +68,77 @@ const getItemById = async (req, res) => {
   }
 };
 
+//POST a new inventory item
+const postNewInventoryItem = async (req, res) => {
+  try {
+    const {
+      item_name,
+      description,
+      category,
+      status,
+      quantity,
+      warehouse_name,
+    } = req.body;
+
+    // Validate data
+    if (
+      !item_name ||
+      !description ||
+      !category ||
+      !status ||
+      quantity === undefined ||
+      !warehouse_name
+    ) {
+      return res.status(400).json({ message: "Missing required fields" });
+    }
+    // Check if quantity is a valid number
+    if (isNaN(parseInt(quantity, 10))) {
+      return res.status(400).json({ message: "Quantity must be a number" });
+    }
+
+    // Look up the warehouse ID based on the warehouse name
+    const warehouse = await knex("warehouses")
+      .where("warehouse_name", warehouse_name)
+      .first();
+    if (!warehouse) {
+      return res.status(400).json({ message: "Warehouse name does not exist" });
+    }
+
+    // Insert new item into the database using the found warehouse ID
+    const [newItemId] = await knex("inventories")
+      .insert({
+        item_name,
+        description,
+        category,
+        status,
+        quantity, // Already validated as a number
+        warehouse_id: warehouse.id, // Use the ID from the warehouse we found above
+      })
+      .returning("id");
+
+    // Retrieve and respond with the inserted item
```

Comment:
Suggestion: we don't need to querry the newly added item. If all validation passed, we can return the item object we used to POST

Reply:
Good call out. I think the resource I was using to learn how to build this was based on postgresql. I'll take this out for our case.